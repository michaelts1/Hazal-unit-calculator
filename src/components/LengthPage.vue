<script lang="ts">
import { Ruler, Unit } from '../types'
import { defineComponent } from 'vue'
import LengthTable from './LengthTable.vue'
import SelectRabi from './SelectRabi.vue'
import { UnitValues } from '../helpers'

export default defineComponent({
	components: {
		LengthTable,
		SelectRabi,
	},

	data() {
		return {
			ruler: 'גרח' as Ruler,
		}
	},

	computed: {
		units(): Unit[] {
			const unitValues = new UnitValues(this.ruler)
			return [
				{ hidden: true, name: 'סנטימטר', value: 0.01 },
				{ hidden: true, name: 'מטר', value: 1 },
				{ hidden: true, name: 'קילומטר', value: 1000 },
				{ hidden: false, name: 'אצבע', value: unitValues.etzba },
				{ hidden: false, name: 'טפח', value: unitValues.tefach },
				{ hidden: false, name: 'סיט', value: unitValues.sit },
				{ hidden: false, name: 'זרת', value: unitValues.zeret },
				{ hidden: false, name: 'אמה', value: unitValues.amah },
				{ hidden: false, name: 'ריס', value: unitValues.ris },
				{ hidden: false, name: 'מיל', value: unitValues.mil },
				{ hidden: false, name: 'פרסה', value: unitValues.parsa },
			]
		},
	},
})
</script>

<template>
	<SelectRabi
		:default-ruler="ruler"
		@ruler-change="newRuler => ruler = newRuler"
	/>

	<h6>שים לב: הסיט מחושב לפי שיטת הרמב"ם</h6>
	<LengthTable :units="units" />
</template>
