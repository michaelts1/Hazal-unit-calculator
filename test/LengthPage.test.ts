import { describe, expect, it } from 'vitest'
import LengthPage from '../src/components/LengthPage.vue'
import { mount } from '@vue/test-utils'
import type { Unit } from '../src/types'

describe('Component LengthPage', () => {
	it('exists', () => {
		expect(LengthPage).toBeTruthy()
	})

	const wrapper = mount(LengthPage)

	it('matches snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('should have the correct values', async () => {
		const current = (wrapper.vm.units as Unit[]).slice(3)
		const expected = [ .02, .08, .16, .24, .48, 128, 960, 3840 ]

		expect(current.map(unit => unit.value)).toEqual(expected)
	})

	it('responds to events emitted by SelectRabi', async () => {
		const SelectRabi = wrapper.findComponent({ name: 'SelectRabi' })

		SelectRabi.vm.$emit('ruler-change', 'חזון_איש')
		expect((wrapper.vm.units as Unit[])[3].value).toBe(.024) // New value
	})
})
