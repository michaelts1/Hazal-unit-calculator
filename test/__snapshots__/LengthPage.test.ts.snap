// Vitest Snapshot v1

exports[`Component LengthPage > matches snapshot 1`] = `
"<label for=\\"ruler-select\\">שיטת מדידה: </label><select id=\\"ruler-select\\">
  <option value=\\"גרח\\"> גר\\"ח (רב חיים נאה) </option>
  <option value=\\"חזון_איש\\"> חזון איש </option>
</select>
<h6>שים לב: הסיט מחושב לפי שיטת הרמב\\"ם</h6>
<table data-v-4279aa88=\\"\\">
  <thead data-v-4279aa88=\\"\\">
    <tr data-v-4279aa88=\\"\\">
      <th data-v-4279aa88=\\"\\">אצבע</th>
      <th data-v-4279aa88=\\"\\">טפח</th>
      <th data-v-4279aa88=\\"\\">סיט</th>
      <th data-v-4279aa88=\\"\\">זרת</th>
      <th data-v-4279aa88=\\"\\">אמה</th>
      <th data-v-4279aa88=\\"\\">ריס</th>
      <th data-v-4279aa88=\\"\\">מיל</th>
      <th data-v-4279aa88=\\"\\">פרסה</th>
    </tr>
  </thead>
  <tbody data-v-4279aa88=\\"\\">
    <tr data-v-4279aa88=\\"\\">
      <td data-v-4279aa88=\\"\\">2 ס\\"מ</td>
      <td data-v-4279aa88=\\"\\">8 ס\\"מ</td>
      <td data-v-4279aa88=\\"\\">16 ס\\"מ</td>
      <td data-v-4279aa88=\\"\\">24 ס\\"מ</td>
      <td data-v-4279aa88=\\"\\">48 ס\\"מ</td>
      <td data-v-4279aa88=\\"\\">128 מטר</td>
      <td data-v-4279aa88=\\"\\">960 מטר</td>
      <td data-v-4279aa88=\\"\\">3.84 ק\\"מ</td>
    </tr>
  </tbody>
</table>
<div class=\\"wrapper-row\\" data-v-6764e49b=\\"\\">
  <div class=\\"wrapper-column\\" data-v-6764e49b=\\"\\"><select class=\\"centered\\" data-v-6764e49b=\\"\\">
      <option value=\\"סנטימטר\\">סנטימטר</option>
      <option value=\\"מטר\\">מטר</option>
      <option value=\\"קילומטר\\">קילומטר</option>
      <option value=\\"אצבע\\">אצבע</option>
      <option value=\\"טפח\\">טפח</option>
      <option value=\\"סיט\\">סיט</option>
      <option value=\\"זרת\\">זרת</option>
      <option value=\\"אמה\\">אמה</option>
      <option value=\\"ריס\\">ריס</option>
      <option value=\\"מיל\\">מיל</option>
      <option value=\\"פרסה\\">פרסה</option>
    </select><input class=\\"conversion-input\\" size=\\"12\\" data-v-519321c2=\\"\\" data-v-6764e49b=\\"\\"></div>
  <div class=\\"wrapper-column\\" data-v-6764e49b=\\"\\"> = </div>
  <div class=\\"wrapper-column\\" data-v-6764e49b=\\"\\"><select class=\\"centered\\" data-v-6764e49b=\\"\\">
      <option value=\\"סנטימטר\\">סנטימטר</option>
      <option value=\\"מטר\\">מטר</option>
      <option value=\\"קילומטר\\">קילומטר</option>
      <option value=\\"אצבע\\">אצבע</option>
      <option value=\\"טפח\\">טפח</option>
      <option value=\\"סיט\\">סיט</option>
      <option value=\\"זרת\\">זרת</option>
      <option value=\\"אמה\\">אמה</option>
      <option value=\\"ריס\\">ריס</option>
      <option value=\\"מיל\\">מיל</option>
      <option value=\\"פרסה\\">פרסה</option>
    </select><input class=\\"conversion-input\\" disabled=\\"\\" size=\\"12\\" data-v-519321c2=\\"\\" data-v-6764e49b=\\"\\"></div>
</div>"
`;
